proposals/readme.adoc

```diff
@@ -1,7 +1,41 @@
 = Proposals
 
-List of proposals and associated links
+List of proposals and associated links.
 
-. link:https://github.com/microprofile/evolution/blob/master/proposals/0001-config.md[MP-0001: Service Configuration]
+This list includes both approved proposals and proposals in the discussion stage. Those proposals which are not yet approved are marked as such.
```

Comment:
yes, it was mostly a decision based on the fact that only Config and Healthcheck have actually been merged to the proposals repo, but I wanted to include the ones that are still in the "PR/Discussion" phase. There's also the JWT proposal which doesn't yet have a PR to link to so that one isn't even listed.

It's not a great solution here, but I couldn't think of a better one at the time :) happy to adjust to any suggestion of a better way though!